07week/ticTacToe/script.js

```diff
@@ -1,46 +1,212 @@
 'use strict';
-
 class TicTacToe extends React.Component {
   constructor(props) {
     super(props);
+    this.state= {
+      symbol: 'O',
+      array: [null, null, null, null, null, null, null, null, null]
+    }
+    this.printSymbol = this.printSymbol.bind(this);
   }
-  getInitialState: function() {
-    return{symbol:"x"}
-  }
-  var Symboll = react.createClass({
-    return
-  })
-  var hello = printSymbol: function(){
-    let i = 0
-    if(i=== 0 || i%2===0){
-      console.log(this.symbol)
-      return (<div data-cell="{i}">this.symbol</div>)
-      i++
-    } else{
-      symbol = "o"
-      console.log(this.symbol)
-      return this.symbol
-      i++
+  printSymbol(event){
+    let newSymbol = this.state.symbol;
+    let newarray = this.state.array;
+    if(newSymbol==='O'){
+      newSymbol='X'
+    } else if (newSymbol==='X'){
+      newSymbol='O'
+    }
+    let index = event.target.getAttribute("data-cell")
+    console.log(newSymbol)
+    if(newarray[index]!== null){
+      return
+    }
+    if(newarray[0]==="X" && newarray[1]==="X" && newarray[2]==="X"){
```

Comment:
While checking for all possible way to win is valid, this approach is needlessly verbose.
You could define all the win conditions in an array, and check against that.